pkg/yamltemplate/filetests/ytt-library/assert-try-to.yml

```diff
@@ -0,0 +1,39 @@
+#@ load("@ytt:assert", "assert")
+#@ load("@ytt:json", "json")
+
+#@ def try_to_fail():
+#@   _, err = assert.try_to(lambda: json.decode("}junk"))
+#@   return err
+#@ end
+
+#@ def try_to_succeed_lambda():
+#@   decoded, err = assert.try_to(lambda: json.decode('{"ytt":"rules"}'))
+#@   if err:
+#@      return err
+#@   end
+#@   return decoded
+#@ end
+
+
+#@ def foo():
+#@   return "foo"
+#@ end
+
+#@ def try_to_succeed_function():
+#@   functionResult, err = assert.try_to(foo)
+#@   if err:
+#@      return err
+#@   end
+#@   return functionResult
+#@ end
+
+test_try_to_fail: #@ try_to_fail()
```

Comment:
lets also add another test that accesses a unknown field like `(None).foo` (results in `NoneType has no .foo field or method` error)